src/index.ts

```diff
@@ -49,6 +47,76 @@ export class PeaqApp extends BaseApp {
     this.eth = new Eth(transport, ethScrambleKey, ethLoadConfig);
   }
 
+  async getAddress(path: BIP32Path, showAddrInDevice = false): Promise<GenericeResponseAddress> {
+    const bip44PathBuffer = this.serializePath(path);
+
+    const p1 = showAddrInDevice ? P1_VALUES.SHOW_ADDRESS_IN_DEVICE : P1_VALUES.ONLY_RETRIEVE;
+
+    try {
+      const responseBuffer = await this.transport.send(this.CLA, this.INS.GET_ADDR, p1, 0, bip44PathBuffer);
+
+      const response = processResponse(responseBuffer);
+
+      const pubKey = response.readBytes(32).toString("hex");
+      const address = response.readBytes(response.length()).toString("ascii");
+
+      return {
+        pubKey,
+        address,
+        return_code: 0x9000,
+        error_message: "No errors",
+      } as GenericeResponseAddress;
+    } catch (e) {
+      throw processErrorResponse(e);
+    }
+  }
+
+  private splitBufferToChunks(message: Buffer, chunkSize: number) {
+    const chunks = [];
+    const buffer = Buffer.from(message);
+
+    for (let i = 0; i < buffer.length; i += chunkSize) {
+      let end = i + chunkSize;
+      if (i > buffer.length) {
+        end = buffer.length;
+      }
+      chunks.push(buffer.subarray(i, end));
+    }
+
+    return chunks;
+  }
+
+  private getSignReqChunks(path: BIP32Path, message: Buffer) {
+    const chunks: Buffer[] = [];
+    const bip44Path = this.serializePath(path);
+
+    const blobLen = Buffer.alloc(2);
+    blobLen.writeUInt16LE(message.length);
+
+    chunks.push(Buffer.concat([bip44Path, blobLen]));
+    chunks.push(...this.splitBufferToChunks(message, this.CHUNK_SIZE));
+
+    return chunks;
+  }
```

Comment:
Are these two functions different to what we do on other chains? They seem pretty generic. If they are indeed, we could migrate/use from ledger-js, adding tests for them there.